test/integration/http_integration.cc

```diff
@@ -275,6 +304,44 @@ void HttpIntegrationTest::useAccessLog(
 
 HttpIntegrationTest::~HttpIntegrationTest() { cleanupUpstreamAndDownstream(); }
 
+void HttpIntegrationTest::initialize() {
+  if (downstream_protocol_ != Http::CodecClient::Type::HTTP3) {
+    return BaseIntegrationTest::initialize();
+  }
+  NiceMock<Server::Configuration::MockTransportSocketFactoryContext> mock_factory_ctx;
+  ON_CALL(mock_factory_ctx, api()).WillByDefault(testing::ReturnRef(*api_));
+
+  quic_transport_socket_factory_ =
+      IntegrationUtil::createQuicClientTransportSocketFactory(mock_factory_ctx, san_to_match_);
```

Comment:
would it be possible to have this either take options or share code from the SSL function, so if we need to change TLS defaults we only have to do it in one place?